[quant][observer] Change exception to warning

hx89 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #23959 [quant][observer] Add histogram observer
* **#25408 [quant][observer] Change exception to warning**

Change exception to warning so that observer can be called with no data and still provide a scale and zero-point.

Differential Revision: [D17116524](https://our.internmc.facebook.com/intern/diff/D17116524/)

Commit:
[quant][observer] Change exception to warning

Change exception to warning so that observer can be called with no data and still provide a scale and zero-point.

Differential Revision: [D17116524](https://our.internmc.facebook.com/intern/diff/D17116524/)

raghuramank100 reviewed (changes requested):
Please see comments, thanks

Commit:
Update on "[quant][observer] Change exception to warning"


Change exception to warning so that observer can be called with no data and still provide a scale and zero-point.

Differential Revision: [D17116524](https://our.internmc.facebook.com/intern/diff/D17116524/)

Commit:
Update on "[quant][observer] Change exception to warning"


Change exception to warning so that observer can be called with no data and still provide a scale and zero-point.

Differential Revision: [D17116524](https://our.internmc.facebook.com/intern/diff/D17116524/)
